Source/Utils/Parser/ParserUtils.H

```diff
@@ -221,31 +137,21 @@ namespace utils::parser
     int queryArrWithParser (const amrex::ParmParse& a_pp, char const * const str, std::vector<T>& val,
                             const int start_ix, const int num_val)
     {
-        // call amrex::ParmParse::query, check if the user specified str.
-        std::vector<std::string> tmp_str_arr;
-        const int is_specified = a_pp.queryarr(str, tmp_str_arr, start_ix, num_val);
+        const int is_specified = queryArrWithParser(a_pp, str, val);
         if (is_specified)
         {
-            // If so, create parser objects and apply them to the values provided by the user.
-            int const n = static_cast<int>(tmp_str_arr.size());
-            val.resize(n);
-            for (int i=0 ; i < n ; i++) {
-                auto parser = makeParser(tmp_str_arr[i], {});
-                if constexpr (std::is_same_v<T, int>) {
-                    val[i] = safeCastToInt(
-                        static_cast<amrex::Real>(std::round(parser.compileHost<0>()())), str);
-                }
-                else {
-                    val[i] = static_cast<T>(parser.compileHost<0>()());
-                }
+            val.erase(val.begin(), val.begin()+start_ix);
+            if (num_val != amrex::ParmParse::LAST) {
+                if (num_val > int(val.size())) { return 0; }
+                val.resize(num_val);
```

Comment:
Could you give a couple of examples of which use cases this logic would cover? For example, what is a user input such that `num_val != amrex::ParmParse::LAST`?